apps/homescreen/test/unit/datastore_test.js

```diff
@@ -0,0 +1,383 @@
+            assert.equal(objectStores[bookmarks.storeName].data, data);
+            done();
+          });
+      });
+
+      test('should emit an event', done => {
+        document.addEventListener(bookmarks.name + '-set', evt => {
+          assert.equal(evt.detail.id, data.id);
+          done();
+        });
+        bookmarks.set(data);
+      });
+    });
+
+    suite('Datastore#remove()', () => {
+      var id = 'abc/';
+
+      setup(() => {
+        attachDBTransaction(bookmarks.storeName);
+      });
+
+      test('should remove data from object store', done => {
+        bookmarks.remove(id)
+          .then(() => {
+            assert.deepEqual(deletedValues, [id]);
+            done();
+          });
+      });
+
+      test('should emit an event', done => {
+        document.addEventListener(bookmarks.name + '-removed', evt => {
+          assert.equal(evt.detail.id, id);
+          done();
+        });
+        bookmarks.remove(id);
+      });
+    });
+
+    suite('Datastore#clear()', () => {
+      test('should remove all data from object store', done => {
+        attachDBTransaction(bookmarks.storeName);
+        bookmarks.clear()
+          .then(() => {
+            done();
+          });
+      });
+
+      test('should emit an event', done => {
+        attachDBTransaction(bookmarks.storeName);
+        document.addEventListener(bookmarks.name + '-cleared', () => {
+          done();
+        });
+        bookmarks.clear();
+      });
+    });
+
+    suite('Datastore#get()', () => {
+      test('should retrieve data from object store', done => {
+        attachDBTransaction(bookmarks.storeName);
+        bookmarks.get('abc/')
+          .then(results => {
+            assert.deepEqual(results, { id: 'abc/', data: {} });
+            done();
+          });
+      });
+    });
+
+    suite('Datastore#getAll()', () => {
+      test('should retrieve data from object store', done => {
+        attachDBTransaction(bookmarks.storeName);
+        bookmarks.getAll()
+          .then(results => {
+            assert.equal(results.length, 2);
+            assert.deepEqual(results[0], { id: 'abc/', data: {} });
+            assert.deepEqual(results[1], { id: 'def/', data: {} });
+            done();
+          });
+      });
+    });
+  });
+
+  suite('Datastore#updateRevision()', () => {
+    test('should set the datastore revision ID in lastRevision', done => {
+      assert.isNull(bookmarks.lastRevision);
+      bookmarks.init().then(() => {
+        bookmarks.updateRevision();
+        assert.equal(bookmarks.lastRevision, bookmarks.datastore.revisionId);
+        done();
+      });
+    });
+
+    test('should update the datastore revision ID in lastRevision', done => {
+      bookmarks.init().then(() => {
+        bookmarks.updateRevision();
+        assert.equal(bookmarks.lastRevision, bookmarks.datastore.revisionId);
+        bookmarks.datastore.revisionId = 'xxx-zzz';
+        bookmarks.init().then(() => {
```

Comment:
I think it'd be better to test this by emitting a change event, rather than calling init again.